### DISCUSSION TITLE
register_next_step_handler missing from AsyncTeleBot

### DISCUSSION THREAD

**Discussion by TornaD-oz** [eternnoir/pyTelegramBotAPI#1827] [Q&A] [answered]
Hi all, register_next_step_handler is missing from Async TeleBot implementation. Is this intended? If so, how do I replicate the same behavior - sending a specific message to user and then calling a specific function with parameters after user sends a text message?

**Comment by coder2020official** [comment, marked as answer]
Use states.

**Reply by TornaD-oz** [reply]
How do I make AsyncTeleBot wait for an answer and save it to state then?

**Reply by coder2020official** [reply]
Look at examples/asynchronous_telebot/custom_states.py

**Reply by TornaD-oz** [reply]
Thanks for the tip! Guess I'll have to rewrite everything to states logic then.
